virtool/caches/db.py

```diff
@@ -136,12 +134,10 @@ async def create(
             "_id": cache_id,
             "created_at": virtool.utils.timestamp(),
             "files": list(),
-            "key": calculate_cache_key(parameters),
+            "key": key,
```

Comment:
Best approach here is a [UNIQUE] constraint on `key` (https://www.postgresqltutorial.com/postgresql-unique-constraint/). We could probably just handle the SQL error in the request handler instead of raising a different exception in `db.create()`.

Reply:
Could use a `unique` index in MongoDB https://docs.mongodb.com/manual/core/index-unique/